src/css/adjustCSS.js

```diff
@@ -16,10 +17,11 @@ function adjustCSS( elem, prop, valueParts, tween ) {
 				return jQuery.css( elem, prop, "" );
 			},
 		initial = currentValue(),
-		unit = valueParts && valueParts[ 3 ] || ( jQuery.cssNumber[ prop ] ? "" : "px" ),
+		unit = valueParts && valueParts[ 3 ] || ( isAutoPx( prop ) ? "px" : "" ),
 
 		// Starting value computation is required for potential unit mismatches
-		initialInUnit = ( jQuery.cssNumber[ prop ] || unit !== "px" && +initial ) &&
+		initialInUnit = elem.nodeType &&
```

Comment:
Without this change, effects tests were failing on non-element objects on an `a` property. I've noticed they were already failing if `a` was changed to anything from `jQuery.cssNumber` and this PR makes almost everything "like from `jQuery.cssNumber`".

The `nodeType` check should most likely already be there, perhaps we should add it for 3.4.0 as well?

Reply:
Oh, ugh. Put another way, we only want to add "px" to property values if they're on DOM elements. I still wish we hadn't supported animating plain objects,.

Reply:
I extracted this change to PR #4061.

Reply:
#4601 landed, this PR rebased so this particular change is no longer a part of this PR.